flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSinkV2.java

```diff
@@ -0,0 +1,330 @@
+@RunWith(Parameterized.class)
+public class TestFlinkIcebergSinkV2 extends TableTestBase {
+  private static final int FORMAT_V2 = 2;
+  private static final TypeInformation<Row> ROW_TYPE_INFO =
+      new RowTypeInfo(SimpleDataUtil.FLINK_SCHEMA.getFieldTypes());
+
+  private final FileFormat format;
+  private final int parallelism;
+  private final boolean partitioned;
+
+  private StreamExecutionEnvironment env;
+  private TestTableLoader tableLoader;
+
+  @Parameterized.Parameters(name = "FileFormat = {0}, Parallelism = {1}, Partitioned={2}")
+  public static Object[][] parameters() {
+    return new Object[][] {
+        new Object[] {"avro", 1, true},
+        new Object[] {"avro", 1, false},
+        new Object[] {"avro", 2, true},
+        new Object[] {"avro", 2, false},
+        new Object[] {"parquet", 1, true},
+        new Object[] {"parquet", 1, false},
+        new Object[] {"parquet", 2, true},
+        new Object[] {"parquet", 2, false}
+    };
+  }
+
+  public TestFlinkIcebergSinkV2(String format, int parallelism, boolean partitioned) {
+    super(FORMAT_V2);
+    this.format = FileFormat.valueOf(format.toUpperCase(Locale.ENGLISH));
+    this.parallelism = parallelism;
+    this.partitioned = partitioned;
+  }
+
+  @Before
+  public void setupTable() throws IOException {
+    this.tableDir = temp.newFolder();
+    this.metadataDir = new File(tableDir, "metadata");
+    Assert.assertTrue(tableDir.delete());
+
+    if (!partitioned) {
+      table = create(SimpleDataUtil.SCHEMA, PartitionSpec.unpartitioned());
+    } else {
+      table = create(SimpleDataUtil.SCHEMA, PartitionSpec.builderFor(SimpleDataUtil.SCHEMA).identity("data").build());
+    }
+
+    table.updateProperties()
+        .set(TableProperties.DEFAULT_FILE_FORMAT, format.name())
+        .commit();
+
+    env = StreamExecutionEnvironment.getExecutionEnvironment()
+        .enableCheckpointing(100L)
+        .setParallelism(parallelism)
+        .setMaxParallelism(parallelism);
+
+    tableLoader = new TestTableLoader(tableDir.getAbsolutePath());
+  }
+
+  private List<Snapshot> findValidSnapshots(Table table) {
+    List<Snapshot> validSnapshots = Lists.newArrayList();
+    for (Snapshot snapshot : table.snapshots()) {
+      if (snapshot.allManifests().stream().anyMatch(m -> snapshot.snapshotId() == m.snapshotId())) {
+        validSnapshots.add(snapshot);
+      }
+    }
+    return validSnapshots;
+  }
+
+  private void testChangeLogs(List<String> equalityFieldColumns,
+                              KeySelector<Row, Row> keySelector,
+                              List<List<Row>> elementsPerCheckpoint,
+                              List<List<Record>> expectedRecordsPerCheckpoint) throws Exception {
+    DataStream<Row> dataStream = env.addSource(new BoundedTestSource<>(elementsPerCheckpoint), ROW_TYPE_INFO);
+
+    // Shuffle by the equality key, so that different operations from the same key could be wrote in order when
+    // executing tasks in parallelism.
```

Comment:
Nit: I think you mean "executing tasks in parallel" rather than "parallelism".

Reply:
Thanks for pointing it out, will address it in next update.